lib/audited/auditor.rb

```diff
@@ -196,10 +196,22 @@ def revision_with(attributes)
 
       def audited_changes
         all_changes = respond_to?(:changes_to_save) ? changes_to_save : changes
-        if audited_options[:only].present?
-          all_changes.slice(*self.class.audited_columns)
-        else
-          all_changes.except(*self.class.non_audited_columns)
+        filtered_changes = \
+          if audited_options[:only].present?
+            all_changes.slice(*self.class.audited_columns)
+          else
+            all_changes.except(*self.class.non_audited_columns)
+          end
+
+        filtered_changes = normalize_enum_changes(filtered_changes)
+        filtered_changes.to_hash
+      end
+
+      def normalize_enum_changes(changes)
+        changes.each do |key, value|
```

Comment:
instead of looping over all the changes, how about looping on the definded enums? in most cases there would be 0-2 enums while changes could be many more.